### PR TITLE
added function for histograms merge

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
Added function to merge histograms

### CODE REVIEW

File: histogram.go [hunk 1]
```diff
@@ -109,14 +109,36 @@ func (h *Histogram) Update(v float64) {
 	h.mu.Unlock()
 }
 
+// Merge merges histograms
+func (h *Histogram) Merge(b *Histogram) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	b.mu.RLock()
```
Comment (tenmozes) [inline histogram.go:117]: does Lock cover RLock as well?

Comment (tenmozes) [reply histogram.go:117]: https://pkg.go.dev/sync#RWMutex.Lock

Lock locks rw for writing. If the lock is already locked for reading or writing, Lock blocks until the lock is available. 

so just Lock is enough

Comment (AndrewChubatiuk) [reply histogram.go:117]: there're locks on different objects - write lock on a target histogram and read lock on src histogram

Comment (tenmozes) [reply histogram.go:117]: oh sorry, makes sence ;)

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by valyala** [comment]
@AndrewChubatiuk , @f41gh7 , @tenmozes , please see the follow-up commit - 7a44715774c25df23f0b0a01bdd3525fdfb3bbc6

**Pull request merged**